code/part-one/tests.js

```diff
@@ -0,0 +1,129 @@
+const chai = require('chai');
+const expect = chai.expect;
+const { hash } = require('./utils/helpers.js');
+const { Block, Blockchain, Transaction } = require('./blockchain.js');
+
+
+// Test transaction data
+const testFromAddress = hash('testFromAddress');
+const testToAddress = hash('testToAddress');
+const testAmount = '10.00';
+const transaction = new Transaction(
+  testFromAddress,
+  testToAddress,
+  testAmount
+);
+
+// Test blockchain data
+const blockchain = new Blockchain();
+const genesisBlock = blockchain.chain[0];
+const newBlock = new Block();
+const miningAddress = hash('miningAddress');
+let timesMined = 0;
+
+// Test block data
+const testHash = '0123TESTHASH';
+const testTransactions = [];
+const testNonce = 0;
+const now = Date.now();
+const block = new Block(now, testTransactions, testHash);
+const testDataHash = hash(
+  testHash + now + JSON.stringify(testTransactions) + testNonce
+);
+
+
+// Transaction tests
+describe('Transaction', () => {
+  it('Should create a transaction with all initial values', done => {
+    expect(transaction.fromAddress).to.equal(testFromAddress);
+    expect(transaction.toAddress).to.equal(testToAddress);
+    expect(transaction.amount).to.equal(testAmount);
+    done();
+  });
+});
+
+// Blockchain tests
+describe('Blockchain', () => {
+  it('Should create a blockchain with a genesis block', done => {
+    expect(blockchain).to.have.property('_createGenesisBlock');
```

Comment:
Are these underscored properties one's we are requiring students to create and use? Seems like an implementation detail that should not be tested.

Reply:
The students will be filling these functions out. They are underscored only because other classes do not utilize them but I still think it's beneficial for students to fill in the functionality

Reply:
But why does it matter how the students implement the API if the API works correctly?